src/exo4 - Dependency injection with Reader/exo4.ts

```diff
@@ -44,8 +43,22 @@ export enum Country {
 //
 // HINT: Take a look at `reader.ask` to access the environment value
 
-export const exclamation: (sentence: string) => Reader<Country, string> =
-  unimplemented();
+export const addPunctuation = (country: Country, sentence: string): string => {
+  switch (country) {
+    case Country.France:
+      return `${sentence} !`;
+    case Country.Spain:
+      return `¡${sentence}!`;
+    case Country.USA:
+      return `${sentence}!`;
+  }
+};
+
+export const exclamation = (sentence: string): Reader<Country, string> => 
+  pipe(
+    ask<Country>(),
+    map(country => addPunctuation(country, sentence))
```

Comment:
In the code, for better readibility purposes (and simpler imports - for instance `map` could be imported from `option` or `either`) we would rather write it like this:
```
reader.ask<Country>(),
reader.map(country => addPunctuation(country, sentence))
```